app/models/dscc_number.rb

```diff
@@ -0,0 +1,19 @@
+class DsccNumber < ActiveRecord::Base
+  has_one :defence_request
+
+  validates_uniqueness_of :number, scope: [:year_and_month]
```

Comment:
Still a race condition - need to have a unique index and then capture the error and retry